dtls/sslconnection.py

```diff
@@ -404,18 +404,19 @@ def __init__(self, sock, keyfile=None, certfile=None,
             try:
                 peer_address = sock.getpeername()
             except socket.error:
-                peer_address = None
-            if server_side:
-                post_init = self._init_server(peer_address)
-            else:
-                post_init = self._init_client(peer_address)
-
-        SSL_set_bio(self._ssl.value, self._rbio.value, self._wbio.value)
-        self._rbio.disown()
-        self._wbio.disown()
-        if post_init:
-            post_init()
-
+                peer_address = None
+            if server_side:
+                post_init = self._init_server(peer_address)
+            else:
+                post_init = self._init_client(peer_address)
+
+        SSL_set_options(self._ssl.value, SSL_OP_NO_QUERY_MTU)
+        DTLS_set_link_mtu(self._ssl.value, 1500)

```

Comment:
I recall you asked about this earlier. I've looked through the OpenSSL code, and the logic around MTU determination sure has changed a lot. I think that just setting this to the 802.3 default won't work, though: there can be many reasons that path MTU might be lower than this. One common scenario is DSL: PPPoE induces an 8-byte overhead, reducing the MTU to 1492 right at the network border. I think that PyDTLS would just fail now with this hardcode in such situations.

In bss_dgram.c I see that on Linux, OpenSSL tries to determine the MTU from sockopt IP_MTU as part of BIO_CTRL_DGRAM_QUERY_MTU. On Windows, this path is not taken, and this query just returns 0. The caller then falls into a different code path, which computes a minimum by subtracting a protocol-dependent overhead from a guess.

So I think we need to trace though this and find out why that's not working on Windows. Would you like to do that with a debug version of the OpenSSL dll's you built?